### PR TITLE
13 docker production environment

### PR DESCRIPTION
**Pull request by foobazdev**
1. Create users docker file for production
2. Refactor docker-compose and env.example directories structure

### CODE REVIEW

File: backend/users/docker/Dockerfile.production [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM node:22-alpine as build
+
+WORKDIR /app
+
+COPY package*.json .
+
+RUN npm install
```
Comment (M-Ziusko) [inline backend/users/docker/Dockerfile.production:7]: May be npm run ci?

File: backend/users/docker/Dockerfile.production [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+RUN npm run build 
+
+
+FROM node:22-alpine as prod 
+
+WORKDIR /app
+
+COPY package*.json .
+
+RUN npm install --production
```
Comment (M-Ziusko) [inline backend/users/docker/Dockerfile.production:22]: Why we actually run npm i twice?

File: backend/users/docker/Dockerfile.production [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+COPY --from=build /app/dist ./dist
+COPY --from=build /app/src/prisma/schema.prisma ./prisma/schema.prisma
+COPY --from=build /app/src/prisma/migrations ./prisma/migrations
+
+RUN npm pkg delete prisma
+
+RUN npx prisma generate
+
+CMD npx prisma migrate deploy && node ./dist/src/main.js
```
Comment (M-Ziusko) [inline backend/users/docker/Dockerfile.production:32]: may be npm run start:prod

File: backend/users/docker/Dockerfile.production (hunk 2 above)
Comment (M-Ziusko) [inline backend/users/docker/Dockerfile.production:22]: Why we can't copy node modules from build?

File: env-prod/prod.env.example
Comment (M-Ziusko) [inline env-prod/prod.env.example:1]: Not a big fan of naming file following way (duplicated env  type in name)

File: package.json [hunk 4]
```diff
@@ -2,8 +2,13 @@
   "scripts": {
     "prepare": "husky",
     "install:all": "npm i && cd backend && npm run install:all",
-    "test": "cd backend/users && npm run test",
+    "test": "npm run test --prefix backend/users",
     "test:e2e": "./scripts/users-service-e2e.sh",
+    "start:dev": "docker compose -f ./env-dev/docker-compose.yml up",
+    "start:prod": "docker compose -f ./env-prod/docker-compose.yml up -d --build",
+    "build:images": "docker compose -f ./env-prod/docker-compose.yml build",
```
Comment (M-Ziusko) [inline package.json:9]: It's only for prod? If true, then rename to build:image:prod

**Review by M-Ziusko — CHANGES_REQUESTED** [review]

File: backend/users/docker/Dockerfile.production (hunk 2 above)
Comment (foobazdev) [reply backend/users/docker/Dockerfile.production:22]: We are installing only production dependencies

File: backend/users/docker/Dockerfile.production (hunk 3 above)
Comment (foobazdev) [reply backend/users/docker/Dockerfile.production:32]: There is no such command

File: backend/users/docker/Dockerfile.production (hunk 2 above)
Comment (foobazdev) [reply backend/users/docker/Dockerfile.production:22]: There are devDependencies  on previous stage

File: env-prod/prod.env.example
Comment (foobazdev) [reply env-prod/prod.env.example:1]: With this "env-" prefix directories properly ordered  when you are listing them

File: backend/users/docker/Dockerfile.production (hunk 1 above)
Comment (foobazdev) [reply backend/users/docker/Dockerfile.production]: Done

File: package.json (hunk 4 above)
Comment (foobazdev) [reply package.json]: done

**Pull request merged**